### PR TITLE
Update to Composer 2

### PR DESCRIPTION
**Pull request by Jellyfrog**
No need to roll our own checks anymore since this is now built into Composer 2
Also limit the update check to once a week, it's a bit aggressive now :)

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by Jellyfrog** [comment]
Examples;
>Composer detected issues in your platform: Your Composer dependencies require a PHP version ">= 7.3.0". You are running 7.2.33-1+ubuntu18.04.1+deb.sury.org+1.
  
>Composer detected issues in your platform: Your Composer dependencies require the following PHP extensions to be installed: zip

**Comment by murrant** [comment]
This looks good, but I haven't tested.
One thing is this will only check when ./daily.sh is run.  Previous code checks on application startup.  Not sure if we care about that.

**Comment by Jellyfrog** [comment]
> This looks good, but I haven't tested.
> One thing is this will only check when ./daily.sh is run. Previous code checks on application startup. Not sure if we care about that.

No, it's in the autoloader. The above messages was taken from the site.

File: doc/General/Updating.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ you can do so by running the following commands:
 ```bash
 cd /opt/librenms
 git pull
-composer install --no-dev
+./scripts/composer_wrapper.php install --no-dev
 ./build-base.php
```
Comment (murrant) [inline doc/General/Updating.md:26]: This line could use an update too :D

File: scripts/composer_wrapper.php [hunk 2]
```diff
@@ -55,8 +55,13 @@
...
 
-    // self-update
-    passthru("$exec self-update -q" . $extra_args);
+    // If older than 1 week, try update
+    if (time() - filemtime($install_dir . '/composer.phar') > 60 * 60 * 24 * 7) {
+        // self-update
+        passthru("$exec self-update --quiet --2" . $extra_args);
+    }
+
+    @touch($install_dir . '/composer.phar');
```
Comment (murrant) [inline scripts/composer_wrapper.php:64]: Shouldn't this be inside the if statement?

**Comment by murrant** [comment]
When testing, composer install results in error output, but daily.sh shows success.

File: scripts/composer_wrapper.php (hunk 2 above)
Comment (Jellyfrog) [reply scripts/composer_wrapper.php]: Yes.

File: doc/General/Updating.md [hunk 3]
```diff
@@ -23,7 +23,7 @@ you can do so by running the following commands:
 cd /opt/librenms
 git pull
 ./scripts/composer_wrapper.php install --no-dev
-./build-base.php
+php includes/sql-schema/update.php
```
Comment (murrant) [inline doc/General/Updating.md:26]: oh, I guess people could still be on an age old version, but at least we could remove the. build-base.php script.

Comment (Jellyfrog) [reply doc/General/Updating.md:26]: I'll do some following up cleanup later

**Comment by Jellyfrog** [comment]
> When testing, composer install results in error output, but daily.sh shows success.

What error? Due to Composer 2?  
That it continues is known and a feature, it's so the daily cleanup runs anyway (we should split those into it's own job a sunny day)

**Comment by Jellyfrog** [comment]
I added a commit to make the script at least exit with the right code, now it should *show* as FAIL (but it will continue)

**Review by murrant — APPROVED** [review]
They dialed back the default behavior to exclude extensions very recently.
Added change to re-enable that.
https://github.com/composer/composer/issues/9412

Testing went well otherwise.

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-69-release-changelog-november-2020/14124/1

**Pull request merged**